From 0a69bb62021e114993c5800f8499374920344dc7 Mon Sep 17 00:00:00 2001
From: Bing Jiang <bing.jiang@amlogic.com>
Date: Mon, 2 Nov 2020 16:02:04 +0800
Subject: [PATCH 1/2] uboot: add set root system [1/1]

PD#SWPL-35835

Problem:
  add set root system

Solution:
  add set root system

Verify:
  SC2_AH212 S4_AP222

Change-Id: I71d09b8a425f151aba55ea83b46a19540104dfbd
Signed-off-by: Bing Jiang <bing.jiang@amlogic.com>
---
 board/amlogic/configs/s4_ap222.h             | 9 +++++++++
 board/amlogic/configs/sc2_ah212.h            | 9 +++++++++
 board/amlogic/defconfigs/s4_ap222_defconfig  | 1 -
 board/amlogic/defconfigs/sc2_ah212_defconfig | 3 ++-
 4 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/board/amlogic/configs/s4_ap222.h b/board/amlogic/configs/s4_ap222.h
index 5aed3ce8af5..5a51b5fe73c 100644
--- a/board/amlogic/configs/s4_ap222.h
+++ b/board/amlogic/configs/s4_ap222.h
@@ -168,6 +168,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 0; then "\
+                    "if test ${active_slot} = _a; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_a;"\
+                    "else if test ${active_slot} = _b; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_b;"\
+                    "else if test ${active_slot} = normal; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system;"\
+                    "fi;fi;fi;"\
+                "fi;"\
                 "if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;"\
                 "setenv loadaddr ${loadaddr_kernel};"\
                 "if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;"\
diff --git a/board/amlogic/configs/sc2_ah212.h b/board/amlogic/configs/sc2_ah212.h
index a1032c26656..a83377f3c41 100644
--- a/board/amlogic/configs/sc2_ah212.h
+++ b/board/amlogic/configs/sc2_ah212.h
@@ -173,6 +173,15 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
+                "if test ${avb2} = 0; then "\
+                    "if test ${active_slot} = _a; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_a;"\
+                    "else if test ${active_slot} = _b; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system_b;"\
+                    "else if test ${active_slot} = normal; then "\
+                        "setenv bootargs ${bootargs} root=/dev/system;"\
+                    "fi;fi;fi;"\
+                "fi;"\
                 "if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;"\
                 "setenv loadaddr ${loadaddr_kernel};"\
                 "if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;"\
diff --git a/board/amlogic/defconfigs/s4_ap222_defconfig b/board/amlogic/defconfigs/s4_ap222_defconfig
index 83edadabf91..51ad4d1ed3f 100644
--- a/board/amlogic/defconfigs/s4_ap222_defconfig
+++ b/board/amlogic/defconfigs/s4_ap222_defconfig
@@ -35,7 +35,6 @@ CONFIG_CMD_BMP=y
 CONFIG_CMD_CACHE=y
 # CONFIG_CMD_MISC is not set
 CONFIG_CMD_MTDPARTS=y
-CONFIG_CMD_BOOTCTOL_AVB=y
 CONFIG_CMD_WATCHDOG=y
 CONFIG_CMD_JTAG=y
 CONFIG_AML_PARTITION=y
diff --git a/board/amlogic/defconfigs/sc2_ah212_defconfig b/board/amlogic/defconfigs/sc2_ah212_defconfig
index 13eb2c3be6c..f555b2daf98 100644
--- a/board/amlogic/defconfigs/sc2_ah212_defconfig
+++ b/board/amlogic/defconfigs/sc2_ah212_defconfig
@@ -36,7 +36,8 @@ CONFIG_CMD_IR=y
 CONFIG_CMD_BMP=y
 CONFIG_CMD_CACHE=y
 # CONFIG_CMD_MISC is not set
-CONFIG_CMD_BOOTCTOL_AVB=y
+# CONFIG_CMD_BOOTCTOL_AVB=y
+# CONFIG_CMD_RSVMEM is not set
 CONFIG_CMD_WATCHDOG=y
 CONFIG_CMD_JTAG=y
 CONFIG_CMD_INI=y
-- 
2.31.1

